### PR TITLE
docs: more info about supported base operating systems

### LINKED ISSUE #1237: Document which OSes are available / supported by provisioners
There is a start in the docs right now but we should list which OSes we support and what are the minimum supported versions.  For instance, we plan to support Debian, but only >=8 (Jessie).

### PR DESCRIPTION
**Pull request by ehazlett**
This documents the supported base OS versions supported by Machine.  It also removes the changelog from being embedded in the docs.

Added note about changing SSH user for provider image.

Closes #1237 

Refs #1204

### CODE REVIEW

**Comment by ehazlett** [comment]
@nathanleclaire @fredlf PTAL

File: docs/index.md [hunk 1]
```diff
@@ -1596,26 +1596,25 @@ Environment variables and default values:
...
-* Improvements and isolation of provisioning functionality, so Machine can
-provision and configure the Docker Engine based on OS detection. For details,
-see [PR #553](https://github.com/docker/machine/issues/553).
+by the cloud provider.
+
+| Operating System           | Version          | Notes                   |
+|----------------------------|------------------|-------------------------|
+| Boot2Docker                | 1.5+             | default for local       |
+| Ubuntu                     | 12.04+           | default for remote      |
+| Rancher                    | 0.3+             |                         |
```
Comment (nathanleclaire) [inline docs/index.md]: Nit: RancherOS

File: docs/index.md [hunk 2]
```diff
@@ -1596,26 +1596,25 @@ Environment variables and default values:
...
+| RedHat Enterprise Linux    | 7.0+             | experimental            |
+| CentOS                     | 7+               | experimental            |
+| Fedora                     | 21+              | experimental            |
+
+If you want to use a different base operating system on a remote provider,
+you will need to select the image accordingly for that provider.  For
+example, on DigitalOcean you would use the `--digitalocean-image` flag.
+For Amazon AWS, you would use the `--amazonec2-ami` flag.
+
+> Note: if you change the base image for a provider you may also need to change
```
Comment (nathanleclaire) [inline docs/index.md]: I'd include an example, e.g. you could use this one:

"as well.  For example, the default Red Hat AMI on EC2 expects the SSH user to be `ec2-user`, so you would have to specify this with `--amazonec2-ssh-user ec2-user`.

**Comment by nathanleclaire** [comment]
Couple nits, mostly looks pretty good.

**Comment by UserTaken** [comment]
Any plans for the `generic` driver to also support Arch Linux and its the next most popular base distro? Related to the `generic` driver is not requiring nopasswd sudo by have an option to be prompted for the password like Ansible has with `--ask-become-pass` or `--ask-sudo-pass`.

File: docs/index.md (hunk 1 above)
Comment (ehazlett) [inline docs/index.md]: Sorry I was going by our provisioner names.  Will update.

**Comment by ehazlett** [comment]
@nathanleclaire updated.  also note: i took out the "experimental" note on Swarm provisioning.  That's been in for a while so I think we can mark it as stable as the rest :)

PTAL

**Comment by nathanleclaire** [comment]
\o/

LGTM

**Comment by fredlf** [comment]
Oops. Missed this one. Sorry about that. 

@ehazlett Mary now owns all OS docs. You'll need to ping her in the future. Technically, this needs two docs maintainers LGTMs before merging.

**Comment by ehazlett** [comment]
@fredlf ah thanks fred -- sorry about that.

**Pull request merged**